src/Microsoft.TemplateEngine.Edge/Settings/TemplatePackageManager.cs

```diff
@@ -216,6 +216,37 @@ public async Task<IEnumerable<ITemplateInfo>> GetTemplatesAsync(ITemplatePackage
             return allTemplates.Where(t => t.MountPointUri == templatePackage.MountPointUri);
         }
 
+        /// <summary>
+        /// Returns managed template package <see cref="IManagedTemplatePackage"/> and containing templates <see cref="ITemplateInfo"/>.
+        /// </summary>
+        /// <param name="packageIdentifier">The template package identifier.</param>
+        /// <param name="packageVersion">The template package version.</param>
+        /// <param name="cancellationToken">A cancellation token to cancel the asynchronous operation.</param>
+        /// <returns>The managed template package and the containing templates.</returns>
+        public async Task<(IManagedTemplatePackage? Package, IEnumerable<ITemplateInfo>? Templates)> GetManagedTemplatePackageAsync(string packageIdentifier, string? packageVersion = null, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            var templatePackages = await GetTemplatePackagesAsync(false, cancellationToken).ConfigureAwait(false);
+            var foundPackage = templatePackages.Select(tp => tp as IManagedTemplatePackage)
+                .FirstOrDefault(tp =>
+                {
+                    if (tp?.Identifier == packageIdentifier)
+                    {
+                        return string.IsNullOrWhiteSpace(packageVersion) || tp.Version == packageVersion;
+                    }
+                    return false;
+                });
+
+            if (foundPackage != null)
+            {
+                var templates = await GetTemplatesAsync(foundPackage, cancellationToken).ConfigureAwait(false);
+
+                return (foundPackage, templates);
+            }
+
+            return (null, null);
```

Comment:
please consider if it's better to return just `null` in this case, or throw the exception.